crates/types/src/consensus.rs

```diff
@@ -263,9 +263,57 @@ impl Default for ConsensusMetricsValue {
 }
 
 impl<TYPES: NodeType> Consensus<TYPES> {
+    /// Constructor.
+    #[allow(clippy::too_many_arguments)]
+    pub fn new(
+        validated_state_map: BTreeMap<TYPES::Time, View<TYPES>>,
+        cur_view: TYPES::Time,
+        last_decided_view: TYPES::Time,
+        saved_leaves: CommitmentMap<Leaf<TYPES>>,
+        saved_payloads: BTreeMap<TYPES::Time, Arc<[u8]>>,
+        locked_view: TYPES::Time,
+        high_qc: QuorumCertificate<TYPES>,
+        metrics: Arc<ConsensusMetricsValue>,
+    ) -> Self {
+        Consensus {
+            validated_state_map,
+            vid_shares: BTreeMap::new(),
+            saved_da_certs: HashMap::new(),
+            cur_view,
+            last_decided_view,
+            saved_leaves,
+            saved_payloads,
+            locked_view,
+            high_qc,
+            metrics,
+            dontuse_decided_upgrade_cert: None,
+            dontuse_formed_upgrade_certificate: None,
+        }
+    }
+
+    /// Get the current view.
+    pub fn cur_view(&self) -> TYPES::Time {
+        self.cur_view
+    }
+
     /// Update the current view.
-    pub fn update_view(&mut self, view_number: TYPES::Time) {
-        self.cur_view = view_number;
+    pub fn update_view_if_new(&mut self, view_number: TYPES::Time) {
```

Comment:
Minor nit: I think the `if_new` is redundant and these could just be `update_view` and `update_high_qc`. This is not a strong opinion, and might just be personal preference so feel free to ignore if you prefer this naming.